modAionImpl/src/org/aion/zero/impl/db/AionBlockStore.java

```diff
@@ -1250,6 +1251,36 @@ public void correctMainChain(AionBlock block, Logger log) {
         }
     }
 
+    /**
+     * Performed before --redo-import to clear side chain blocks and reset the index.
+     *
+     * @param block the block that will be re-imported and should not be removed from the database
+     */
+    public void redoIndexWithoutSideChains(AionBlock block) {
+        lock.writeLock().lock();
+
+        try {
+            AionBlock currentBlock = block;
+
+            if (currentBlock != null) {
+                byte[] currentHash = block.getHash();
+                List<BlockInfo> level = getBlockInfoForLevel(currentBlock.getNumber());
+
+                // delete all the side-chain blocks
+                for (BlockInfo blockInfo : level) {
+                    if (!Arrays.equals(currentHash, blockInfo.getHash())) {
+                        blocks.delete(blockInfo.getHash());
+                    }
+                }
+
+                // replace all the block info with empty list
+                index.set(block.getNumber(), Collections.emptyList());
```

Comment:
Looks like it only reset the index db at the reimport point level, where is the logic to reset the level between the reimport point to the current index DB height?

Reply:
before the block import, the index is cleared and side-chains are pruned, then the block is imported which sets the correct index value

Reply:
the index is corrected for each block that is imported, it does not modify the database for blocks predating the first block to re-import

Reply:
Okay. understood.